### PR TITLE
refactor(event): optimize event handle function 

### PR DESCRIPTION
**Pull request by wanglei4687**
Signed-off-by: wanglei4687 [wanglei4687@gmail.com](mailto:wanglei4687@gmail.com)

**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #

**What this PR does / why we need it:**

1. refactor event 
2. add actor interface for handler event and so on
3. add  ObjStateTransFunc for watching state and hanldle state transfer event

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by wanglei4687** [comment]
@mergify update

**Comment by wanglei4687** [comment]
@mergify update

File: pkg/controllers/matrixone/matrixonecluster_controller.go [hunk 1]
```diff
@@ -37,6 +39,8 @@ type ClusterReconciler struct {
 	Log           logr.Logger
 	ReconcileWait time.Duration
 	Recorder      record.EventRecorder
+	Actor         actor.ObjActorFunc
+	StateHandler  state.ObjStateTransFunc
```
Comment (aylei) [inline pkg/controllers/matrixone/matrixonecluster_controller.go:43]: where are these fields initialized?

File: pkg/state/state.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package state
+
+//ObjStateTransFunc interface for handle state transfer
+// like pod / cluster_infra.
+type ObjStateTransFunc interface{}
```
Comment (aylei) [inline pkg/state/state.go:19]: Perhaps we need a specific interface definition

**Comment by wanglei4687** [comment]
close it because of  new matrixone operator design

**Pull request closed without merging**